catalog/core/catalog-core-standardframework/src/main/java/org/codice/ddf/catalog/resource/download/impl/ResourceDownload.java

```diff
@@ -0,0 +1,169 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.resource.download.impl;
+
+import java.io.IOException;
+
+import javax.management.InstanceAlreadyExistsException;
+import javax.management.InstanceNotFoundException;
+import javax.management.MBeanException;
+import javax.management.MBeanRegistrationException;
+import javax.management.MBeanServer;
+import javax.management.MalformedObjectNameException;
+import javax.management.NotCompliantMBeanException;
+import javax.management.ObjectName;
+import javax.management.StandardMBean;
+import javax.ws.rs.core.Response.Status;
+
+import org.codice.ddf.catalog.resource.cache.ResourceCacheServiceMBean;
+import org.codice.ddf.catalog.resource.download.DownloadToCacheException;
+import org.codice.ddf.catalog.resource.download.ResourceDownloadMBean;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.operation.impl.ResourceRequestById;
+import ddf.catalog.resource.ResourceNotFoundException;
+import ddf.catalog.resource.ResourceNotSupportedException;
+
+
+public class ResourceDownload implements ResourceDownloadMBean {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ResourceDownload.class);
+
+    private static final String CLASS_NAME = ResourceDownload.class.getName();
+
+    private static final String ERROR_MESSAGE_TEMPLATE = "Unable to download the product associated with metacard [%s] from source [%s] to the product cache.";
```

Comment:
Isn't the product always a file where-as the resource may be data within the metacard itself?